module/zfs/zcp_get.c

```diff
@@ -548,7 +479,7 @@ get_zap_prop(lua_State *state, dsl_dataset_t *ds, zfs_prop_t zfs_prop)
 		    1, &numval, setpoint);
 
 		/* Fill in temporary value for prop, if applicable */
-		(void) get_temporary_prop(ds, zfs_prop, &numval, setpoint);
+		(void) zfs_get_temporary_prop(ds, zfs_prop, &numval, setpoint);
```

Comment:
Alternately, how about this to align the code with the quota property handling in `zcp_get_prop()`.    The only potential user space consumer would be ztest and I didn't see a test case which would hit this.

```diff
diff --git a/module/zfs/zcp_get.c b/module/zfs/zcp_get.c
index 0a5f0b824..ccef9d9f4 100644
--- a/module/zfs/zcp_get.c
+++ b/module/zfs/zcp_get.c
@@ -547,8 +547,14 @@ get_zap_prop(lua_State *state, dsl_dataset_t *ds, zfs_prop_t 
                error = dsl_prop_get_ds(ds, prop_name, sizeof (numval),
                    1, &numval, setpoint);
 
-               /* Fill in temorary value for prop, if applicable */
-               (void) get_temporary_prop(ds, zfs_prop, &numval, setpoint);
+#ifdef _KERNEL
+               /* Fill in temporary value for prop, if applicable */
+               (void) zfs_get_temporary_prop(ds, zfs_prop, &numval, setpoint);
+#else
+                return (luaL_error(state,
+                    "temporary properties only supported in kernel mode",
+                    prop_name));
+#endif
 
                /* Push value to lua stack */
                if (prop_type == PROP_TYPE_INDEX) {
```